ui/src/app/shared/template-resolution.ts

```diff
@@ -0,0 +1,77 @@
+import * as models from '../../models';
+import {ResourceScope, WorkflowStep} from '../../models';
+
+export function getResolvedTemplates(workflow: models.Workflow, node: models.NodeStatus): models.Template {
```

Comment:
this function has a lot of logic in it - so we should unit test it, but it is also a lift-and-shift, so not today